src/main/java/carpet/CarpetSettings.java

```diff
@@ -739,4 +744,61 @@ public static void changeSpawnSize(int size)
             category = {SURVIVAL, CLIENT}
     )
     public static boolean cleanLogs = false;
+
+    public static class StructureBlockLimitValidator extends Validator<Integer> {
+
+        @Override public Integer validate(ServerCommandSource source, ParsedRule<Integer> currentRule, Integer newValue, String string) {
+            return (newValue >= 46) ? newValue : null;
+        }
+
+        @Override
+        public String description() {
+            return "You have to choose a value greater or equal to 46";
+        }
+    }
+    @Rule(
+            desc = "Customizable structure block limit of each axis",
+            extra = {"WARNING: Needs to be permanent for correct loading.",
+                    "Setting 'structureBlockIgnored' to air is recommended",
+                    "when saving massive structures.",
+                    "Required on client of player editing the Structure Block.",
+                    "'structureBlockOutlineDistance' may be required for",
+                    "correct rendering of long structures."},
+            options = {"48", "96", "192", "256"},
+            category = {CREATIVE, CLIENT},
+            validate = StructureBlockLimitValidator.class,
+            strict = false
+    )
+    public static int structureBlockLimit = 48;
+
+    public static class StructureBlockIgnoredValidator extends Validator<String> {
+
+        @Override
+        public String validate(ServerCommandSource source, ParsedRule<String> currentRule, String newValue, String string) {
+            if (!Registry.BLOCK.getIds().contains(Identifier.tryParse(newValue))) {
+                Messenger.m(source, "Unknown block '" + newValue + "'.");
+                return null;
+            }
+            structureBlockIgnoredBlock = Registry.BLOCK.get(Identifier.tryParse(newValue));
```

Comment:
You could simplify the validator to prevent polling the registry twice with something like (note: written in browser and missing imports):

```suggestion
            Optional<Block> ignoredBlock = Registry.BLOCK.getOptional(Identifier.tryParse(newValue));
            if (ignoredBlock.isEmpty()) {
                Messenger.m(source, "Unknown block '" + newValue + "'.");
                return null;
            }
            structureBlockIgnoredBlock = ignoredBlock.get();
```

Reply:
I woudn't be worried about polling registries twice since its just check when changing the setting, but your code is just less hacky (using appropriate contract of the registry.